compose.py

```diff
@@ -0,0 +1,268 @@
+                             text=True)
+    fail_if(p_build.returncode != 0,
+            'docker inspect of upstream docker image build sha failed')
+
+    argmap = {
+        'NVIDIA_BUILD_REF': p_sha.stdout.rstrip(),
+        'NVIDIA_BUILD_ID': p_build.stdout.rstrip(),
+        'TRITON_VERSION': version,
+        'TRITON_CONTAINER_VERSION': container_version,
+    }
+    return argmap
+
+
+# Install dependencies and run entrypoint script
+def end_gpu_dockerfile(ddir, dockerfile_name, argmap, backends, endpoint):
+    import build
+    df = build.dockerfile_add_installation_linux(argmap, backends, endpoint)
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+
+### Create container with docker build
+def build_docker_image(ddir, dockerfile_name, container_name):
+    p = subprocess.Popen(['docker', 'build', '-t', container_name, '-f', \
+        os.path.join(ddir, dockerfile_name), '.'])
+    p.wait()
+    fail_if(p.returncode != 0, 'docker build {} failed'.format(container_name))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    group_qv = parser.add_mutually_exclusive_group()
+    group_qv.add_argument('-q',
+                          '--quiet',
+                          action="store_true",
+                          required=False,
+                          help='Disable console output.')
+    group_qv.add_argument('-v',
+                          '--verbose',
+                          action="store_true",
+                          required=False,
+                          help='Enable verbose output.')
+    parser.add_argument(
+        '--output-name',
+        type=str,
+        required=False,
+        help='Name for the generated Docker image. Default is "tritonserver".')
+    parser.add_argument(
+        '--work-dir',
+        type=str,
+        required=False,
+        help=
+        'Generated dockerfiles are placed here. Default to current directory.')
+    parser.add_argument(
+        '--upstream-container-version',
+        type=str,
+        required=False,
+        help=
+        'The version to use for the generated Docker image. If not specified the container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument('--enable-gpu',
+                        action="store_true",
+                        required=False,
+                        help='Generate a Triton image that supports GPU')
+    parser.add_argument(
+        '--backend',
+        action='append',
+        required=False,
+        help=
+        'Include <backend-name> in the generated Docker image. The flag may be specified multiple times.'
+    )
+    parser.add_argument(
+        '--repoagent',
+        action='append',
+        required=False,
+        help=
+        'Include <repoagent-name> in the generated Docker image. The flag may be specified multiple times.'
+    )
+    parser.add_argument(
```

Comment:
Don't add --endpoint flag because what endpoints are included in Triton cannot be controlled by compose. For sagemaker you need to look into the server image and see if the /usr/bin/serve and the do the sagemaker setup correctly for composed image (this code will be in end_gpu_dockerfile)